### PR TITLE
Add default value `memory` in `KrylovSolver`

### PR DESCRIPTION
**Pull request by tmigot**
Connected to https://github.com/JuliaSmoothOptimizers/LLSModels.jl/pull/35

### CODE REVIEW

**Review by dpo — CHANGES_REQUESTED** [review]

File: src/krylov_solvers.jl [hunk 1]
```diff
@@ -357,7 +357,7 @@ mutable struct DqgmresSolver{T,S} <: KrylovSolver{T,S}
     return solver
   end
 
-  function DqgmresSolver(A, b, memory)
+  function DqgmresSolver(A, b, memory = 20)
```
[the remark this answers is not in the document]
Comment (tmigot) [reply src/krylov_solvers.jl:360]: Thanks for the review. Should I do the same for the 4 solvers?

Should I also change the default parameter in the functions ?https://github.com/JuliaSmoothOptimizers/Krylov.jl/blob/6d28d54aaeda21857a6f6c871686b3eb7cc95f65/src/diom.jl#L38

[the remark this answers is not in the document]
Comment (dpo) [reply src/krylov_solvers.jl:360]: > Thanks for the review. Should I do the same for the 4 solvers?

Yes, please.

> Should I also change the default parameter in the functions ?

Yes I think so. @amontoison ?

**Comment by geoffroyleconte** [comment]
Hi, thanks I will also benefit from this change! Can you do it for GPMR as well please? And maybe inform the user of the default values in the doc?

File: src/krylov_solvers.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (amontoison) [reply src/krylov_solvers.jl:360]: I propose to directly add `memory = min(20, n)` [here](https://github.com/JuliaSmoothOptimizers/Krylov.jl/blob/main/src/krylov_solvers.jl#L34) by default.

[the remark this answers is not in the document]
Comment (amontoison) [reply src/krylov_solvers.jl:360]: We just need to explain that `memory` cannot exceed the size of the linear systems.

[the remark this answers is not in the document]
Comment (amontoison) [reply src/krylov_solvers.jl:360]: It's relevant if we solve small systems (size < 20) but it's also a safeguard if the user gives an inappropriate value.

**Review by amontoison — APPROVED** [review]

File: src/krylov_solvers.jl [hunk 2]
```diff
@@ -357,9 +357,10 @@ Type for storing the vectors required by the in-place version of DQGMRES.
 The outer constructors
 
     solver = DqgmresSolver(n, m, memory, S)
-    solver = DqgmresSolver(A, b, memory)
+    solver = DqgmresSolver(A, b, memory = 20)
 
 may be used in order to create these vectors.
+`memory` is set to `n` if the given value is larger than `n`.
```
Comment (dpo) [inline src/krylov_solvers.jl:363]:
```suggestion
`memory` is set to `n` if the value given is larger than `n`.
```

**Review by dpo — CHANGES_REQUESTED** [review]

**Review by geoffroyleconte — APPROVED** [review]

**Comment by dpo** [comment]
Thank you.

**Pull request merged**